mne/time_frequency/csd.py

```diff
@@ -1,56 +1,512 @@
+                             'length of fmax.')
+
+        # Sum across each frequency bin
+        n_bins = len(fmin_inds)
+        new_data = np.zeros((self._data.shape[0], n_bins),
+                            dtype=self._data.dtype)
+        new_frequencies = []
+        for i, (min_ind, max_ind) in enumerate(zip(fmin_inds, fmax_inds)):
+            new_data[:, i] = self._data[:, min_ind:max_ind].sum(axis=1)
+            new_frequencies.append(self.frequencies[min_ind:max_ind])
+
+        csd_out = CrossSpectralDensity(data=new_data, ch_names=self.ch_names,
+                                       tmin=self.tmin, tmax=self.tmax,
+                                       frequencies=new_frequencies,
+                                       n_fft=self.n_fft)
+        return csd_out
+
+    def mean(self, fmin=None, fmax=None):
+        """Calculate the mean CSD in the given frequency range(s).
+
+        Parameters
+        ----------
+        fmin : float | list of float | None
+            Lower bound of the frequency range in Hertz. Defaults to the lowest
+            frequency available. When a list of frequencies is given, these are
+            used as the lower bounds (inclusive) of frequency bins and the mean
+            is taken for each bin.
+        fmax : float | list of float | None
+            Upper bound of the frequency range in Hertz. Defaults to the
+            highest frequency available. When a list of frequencies is given,
+            these are used as the upper bounds (inclusive) of frequency bins
+            and the mean is taken for each bin.
+
+        Returns
+        -------
+        csd : Instance of CrossSpectralDensity
+            The CSD matrix, averaged across the given frequency range(s).
+        """
+        csd = self.sum(fmin, fmax)
+        for i, f in enumerate(csd.frequencies):
+            csd._data[:, i] /= len(f)
+        return csd
+
+    def _get_frequency_index(self, freq):
+        """Find the index of the given frequency in ``self.frequencies``.
+
+        If the exact given frequency is not available, the nearest frequencies
+        will be chosen, up to a difference of 1 Hertz.
+
+        Parameters
+        ----------
+        freq : float
+            The frequency to find the index for
+
+        Returns
+        -------
+        index : int
+            The index of the frequency nearest to the requested frequency.
+        """
+        if self.is_sum:
+            raise ValueError('This CSD object represents a mean across '
+                             'frequencies. Cannot select a specific '
+                             'frequency.')
+
+        distance = np.abs(np.asarray(self.frequencies) - freq)
+        index = np.argmin(distance)
+        min_dist = distance[index]
+        if min_dist > 1:
+            raise IndexError('Frequency %f is not available.' % freq)
+        return index
+
+    def pick_frequency(self, freq=None, index=None):
```

Comment:
do you really need 2 parameters. You could write:

pick_frequency(self, inst.frequencies[index])

it's still one line and reduces the API complexity.

Reply:
A common use case to to average across frequencies. The `frequencies` field of the resulting CSD object then contains frequency-bins (as tuples). Then, indexing by number is more natural.